### PR TITLE
url extender and unit test

### PR DESCRIPTION
**Pull request by AyAggarwal**

### CODE REVIEW

File: src/test/java/com/algorand/algosdk/transaction/TestTransaction.java [hunk 1]
```diff
@@ -211,6 +212,21 @@ public void testAssetParamsValidation() throws Exception
             .hasMessageContaining("asset metadataHash '" +  badMetadataHash  + "' is not base64 encoded");
 
         assertThatThrownBy(() -> new AssetParams(
+                BigInteger.valueOf(100),
+                3,
+                false,
+                "tst",
+                "testcoin",
+                "websitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsitewebsite",
```
Comment (jdtzmn) [inline src/test/java/com/algorand/algosdk/transaction/TestTransaction.java:220]: This is significantly longer than 96 characters. Additionally, I'd recommend using something more concise and self-explanatory such as `'a'.repeat(97)`.

**Review by jdtzmn — CHANGES_REQUESTED** [review]

File: src/test/java/com/algorand/algosdk/transaction/TestTransaction.java (hunk 1 above)
Comment (AyAggarwal) [reply src/test/java/com/algorand/algosdk/transaction/TestTransaction.java:220]: good idea

**Review by jasonpaulos — COMMENTED** [review]
This looks good, just one suggestion. Could you add a test that verifies a 96 character URL does not throw an error?

**Comment by AyAggarwal** [comment]
> This looks good, just one suggestion. Could you add a test that verifies a 96 character URL does not throw an error?

yes, I'll make the passing case have a url of length 96

File: src/test/java/com/algorand/algosdk/transaction/TestTransaction.java [hunk 2]
```diff
@@ -138,6 +138,7 @@ private void createAssetTest(int numDecimal, String goldenString) throws Excepti
         Address freeze = addr;
         Address clawback = addr;
         String metadataHash = "fACPO4nRgO55j1ndAK3W6Sgc4APkcyFh";
+        String url_96 = new String(new char[96]).replace("\0", "w");;
```
Comment (jdtzmn) [inline src/test/java/com/algorand/algosdk/transaction/TestTransaction.java:141]: Extra semicolon?

File: src/test/java/com/algorand/algosdk/transaction/TestTransaction.java [hunk 3]
```diff
@@ -194,6 +195,8 @@ public void testAssetParamsValidation() throws Exception
         Address clawback = addr;
         String badMetadataHash = "fACPO4nRgO55j1ndAK3W6Sgc4APkcyF!";
         String tooLongMetadataHash = "fACPO4nRgO55j1ndAK3W6Sgc4APkcyFhfACPO4nRgO55j1ndAK3W6Sgc4APkcyFh";
+        String normalMetadataHash = "fACPO4nRgO55j1ndAK3W6Sgc4APkcyFhfACPO4";
+        String url_100 = new String(new char[100]).replace("\0", "w");;
```
Comment (jdtzmn) [inline src/test/java/com/algorand/algosdk/transaction/TestTransaction.java:199]: Extra semicolon?

**Review by jdtzmn — CHANGES_REQUESTED** [review]

**Pull request merged**